packages/next-swc/crates/next-core/src/app_source.rs

```diff
@@ -464,14 +464,15 @@ async fn create_app_page_source_for_route(
     fallback_page: DevHtmlAssetVc,
     intermediate_output_path_root: FileSystemPathVc,
 ) -> Result<ContentSourceVc> {
-    let pathname_vc = StringVc::cell(pathname.to_string());
+    let pathname = pathname.to_string().replace("%5F", "_");
```

Comment:
This should happen in app_structure instead. Also, should this only replace %5F to underscore if at the very beginning of the file name, or anywhere? Does this work with all URL escapes? Please add a comment to explain these cases.

Reply:
Currently it replaces all, not only beginning: https://github.com/vercel/next.js/blob/c390c1662bc79e12cf7c037dcb382ef5ead6e492/packages/next/src/build/entries.ts#L119